aten/src/ATen/core/dispatch/DispatchKeyExtractor.h

```diff
@@ -11,34 +11,32 @@ namespace c10 {
 
 namespace impl {
 
-// Take a DispatchKeySet for a Tensor, and combine it with the current thread
-// local valid (implemented) and enabled (not implemented) DispatchKeySets
-// to determine what the actual dispatch DispatchKey should be.  Unlike
-// Tensor::key_set(), the value of this on a tensor can change depending
-// on TLS.
-//
-// NB: I didn't make this take a Tensor to avoid header include shenanigans.
+// Take a DispatchKeySet for a Tensor and determine the actual dispatch
+// DispatchKey should be, taking into account TLS, and skipping backends which
+// fall through.
 //
-// TODO: I'm not sure if this should live in this header or not; the operant
-// question is whether or not we have access to all the relevant TLS at this
-// point.
+// Unlike Tensor::key_set(), the value of this on a tensor can change depending
+// on TLS.
 static inline DispatchKey dispatchTypeId(
-    DispatchKeySet ts,
-    // This argument is used to mask out fallthrough keys, so we don't
-    // consider them for dispatch.  It's taken in the weird "inverted" form so
-    // that we can do it in one instruction (x & y) rather than two (x & ~y).
-    // These exclusions are NOT tracked in the TLS, but must be applied AFTER
-    // TLS, which is why you have to pass them in to this function (as opposed
-    // to just applying it to the input 'ts').
-    DispatchKeySet non_fallthrough_mask
+    DispatchKeySet ks,
+    // Backends without fallthrough are the only backends we should consider
+    // for dispatch; backends that do have fallthrough should be skipped
+    // under consideration.  These excluded backends are NOT tracked in the TLS,
+    // but must be applied AFTER TLS (since the backend may have been introduced
+    // for consideration by the included TLS), which is why you have to pass
+    // them in to this function (as opposed to just applying it to the input
+    // 'ts').
+    DispatchKeySet backendsWithoutFallthrough
```

Comment:
New wording:

```
      // The key mask lets us eliminate (by zero entries) keys which should not
      // be considered for dispatch.  There is one case when we use this:
      // if there is no operator registered for a backend whose fallback behavior
      // is to fallthrough, we eliminate that backend from consideration (since
      // we want to "fallthrough" to the next valid key.)
      //
      // These excluded backends are NOT tracked in the TLS, but must be applied
      // AFTER TLS (since the backend may have been introduced for consideration
      // by the included TLS), which is why you have to pass them in to this
      // function (as opposed to just applying it to the input 'ks').
      DispatchKeySet key_mask
```